src/main/java/graphql/normalized/NormalizedQueryTreeFactory.java

```diff
@@ -113,34 +144,301 @@ private void fixUpParentReference(NormalizedField rootNormalizedField) {
+        for (NormalizedField nf : normalizedFields) {
+            GraphQLFieldDefinition nfFieldDefinition = nf.getOneFieldDefinition(schema);
+            // same field name
+            if (!nfFieldDefinition.getName().equals(fieldDefinition.getName())) {
+                continue;
+            }
+            // same type
+            if (!simplePrint(nfFieldDefinition.getType()).equals(simplePrint(fieldDefinition.getType()))) {
+                continue;
+            }
+            // same arguments
+            if (!sameArguments(nf.getAstArguments(), arguments)) {
+                continue;
+            }
+            return nf;
+        }
+        return null;
+    }
+
+    // copied from graphql.validation.rules.OverlappingFieldsCanBeMerged
+    private boolean sameArguments(List<Argument> arguments1, List<Argument> arguments2) {
+        if (arguments1.size() != arguments2.size()) {
+            return false;
+        }
+        for (Argument argument : arguments1) {
+            Argument matchedArgument = findArgumentByName(argument.getName(), arguments2);
+            if (matchedArgument == null) {
+                return false;
+            }
+            if (!sameValue(argument.getValue(), matchedArgument.getValue())) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private Argument findArgumentByName(String name, List<Argument> arguments) {
+        for (Argument argument : arguments) {
+            if (argument.getName().equals(name)) {
+                return argument;
+            }
+        }
+        return null;
+    }
+
+    private boolean sameValue(Value value1, Value value2) {
+        if (value1 == null && value2 == null) {
+            return true;
+        }
+        if (value1 == null) {
+            return false;
+        }
+        if (value2 == null) {
+            return false;
+        }
+        return new AstComparator().isEqual(value1, value2);
+    }
+
+
+    private GraphQLFieldDefinition getFieldDefinition(GraphQLSchema schema, GraphQLObjectType objectType, String fieldName) {
+        GraphQLFieldDefinition fieldDefinition;
+        if (fieldName.equals(schema.getIntrospectionTypenameFieldDefinition().getName())) {
+            fieldDefinition = schema.getIntrospectionTypenameFieldDefinition();
+        } else {
+            if (fieldName.equals(schema.getIntrospectionSchemaFieldDefinition().getName())) {
+                fieldDefinition = schema.getIntrospectionSchemaFieldDefinition();
+            } else if (fieldName.equals(schema.getIntrospectionTypeFieldDefinition().getName())) {
+                fieldDefinition = schema.getIntrospectionTypeFieldDefinition();
+            } else {
+                fieldDefinition = assertNotNull(objectType.getField(fieldName), () -> String.format("no field %s found for type %s", fieldName, objectType));
+            }
+        }
+        return fieldDefinition;
+    }
```

Comment:
`graphql.introspection.Introspection#getFieldDef` does this already

Reply:
thanks!